Upgrade to version 3.0

nickjcroucher opened the pull request:
Happy new year! Addresses most of the improvements discussed in https://github.com/nickjcroucher/gubbins/issues/1. CI switched to Github Actions and currently passing all tests on Linux and OSX. Just restructuring of the code and packaging to address, which seems easier to deal with post-PR.

nickjcroucher committed:
Modify to use pyjar

nickjcroucher committed:
Integrate pyjar into reconstruction

nickjcroucher committed:
Enable mar

nickjcroucher committed:
Remove unnecessary pyjar code

nickjcroucher committed:
Improve efficiency of alignment processing

nickjcroucher committed:
Extract function for reconstructing individual columns

nickjcroucher committed:
Tidy up column reconstruction

nickjcroucher committed:
Convert base reconstruction to pool

nickjcroucher committed:
Prepare for sharedmem implementation

nickjcroucher committed:
Pass through number of threads to use to pyjar

nickjcroucher committed:
Parallelisation used shared memory

nickjcroucher committed:
Improve parallelisation job definition

nickjcroucher committed:
Improve phylogeny SNP detection

nickjcroucher committed:
Changed access to shared memory object

nickjcroucher committed:
Improve efficiency of new alignment creation

nickjcroucher committed:
Heuristic for faster reconstruction

nickjcroucher committed:
Tidy up parallelisation code

nickjcroucher committed:
Enable use of iqtree

nickjcroucher committed:
Enable ASR with fasttree

nickjcroucher committed:
Add validation for parameter value extraction

nickjcroucher committed:
Controlled parameterisation of Fasttree

nickjcroucher committed:
Enable use of rapidnj

nickjcroucher committed:
Enable use of raxml to fit models

nickjcroucher committed:
Enable model fitting with fasttree

nickjcroucher committed:
Reorganise help message

nickjcroucher committed:
More flexible algorithm selection

nickjcroucher committed:
Added flexibility to Fasttree

nickjcroucher committed:
Allow for use of different substitution models with fasttree

nickjcroucher committed:
Altered model options for raxml

nickjcroucher committed:
Fix RAxML naming clash

nickjcroucher committed:
Update rapidnj object

nickjcroucher committed:
Allow for selection of initial tree builder

nickjcroucher committed:
Added option to use a starting star phylogeny

nickjcroucher committed:
Enable pairwise comparisons

nickjcroucher committed:
Add nucleotide substitution models

nickjcroucher committed:
Allow more flexible use of substitution models

nickjcroucher committed:
Process input arguments sensibly

nickjcroucher committed:
Update tree building command line options

nickjcroucher committed:
Reconstruct sequences on optimised tree

nickjcroucher committed:
Use unrooted tree for IQtree ASR

nickjcroucher committed:
Tidy up unrooting function

nickjcroucher committed:
Fix for pyjar

nickjcroucher committed:
Minor code tidying

nickjcroucher committed:
Shared memory for base positions

nickjcroucher committed:
Improved use of numpy data structures

nickjcroucher committed:
Merge pull request #2 from nickjcroucher/numpy_base_patterns

Numpy base patterns

nickjcroucher committed:
Merge pull request #4 from nickjcroucher/numpy_base_patterns

Merge in use of numpy shared memory objects.

nickjcroucher committed:
Simplify numpy data processing

nickjcroucher committed:
Merge pull request #5 from nickjcroucher/numpy_base_patterns

Simplify numpy data processing

nickjcroucher committed:
Tidy up parallelisation

nickjcroucher committed:
Enable greater flexibility in model selection

nickjcroucher committed:
Allow input of custom models

nickjcroucher committed:
Allow extra arguments to be passed to algorithms

nickjcroucher committed:
Added bootstrapping for rapidnj

nickjcroucher committed:
Add raxml bootstrapping

nickjcroucher committed:
Add iqtree bootstrap

nickjcroucher committed:
Tidy up raxml bootstraps

nickjcroucher committed:
Rapidnj bootstrap tidied up

nickjcroucher committed:
Add Fasttree bootstrapping

nickjcroucher committed:
Add SH tests for ML methods

nickjcroucher committed:
Tidy up iqtree SH tests

nickjcroucher committed:
Add input argument processing

nickjcroucher committed:
Tidy up shared memory objects

nickjcroucher committed:
Tidy up input options

nickjcroucher committed:
Add installation requirements needed for shared memory objects

nickjcroucher committed:
Increase python version

nickjcroucher committed:
Further checking of input arguments

nickjcroucher committed:
Added in raxmlng as a tree builder

nickjcroucher committed:
Enable sequence reconstruction with RAxML-NG

nickjcroucher committed:
Enable sequence reconstruction with RAxML-NG

nickjcroucher committed:
Enable bootstraps with RAxML-NG

nickjcroucher committed:
Bootstrap annotation using RAxML-NG as a utility

nickjcroucher committed:
Add transfer bootstrap expectation

nickjcroucher committed:
Simplify alignment naming

nickjcroucher committed:
Simplify reconstruction filename code

nickjcroucher committed:
Simplify bootstrap command

nickjcroucher committed:
Tidy up pyjar code

nickjcroucher committed:
Add methods logging

nickjcroucher committed:
Minor fixes to file names

nickjcroucher committed:
Allow for a different model fitter

nickjcroucher committed:
Simplify algorithm selection code

nickjcroucher committed:
Merge branch 'master' into pyjar

nickjcroucher committed:
Merge pull request #6 from nickjcroucher/pyjar

Changes to code for enhancement #1

nickjcroucher committed:
Merge branch 'master' of https://github.com/sanger-pathogens/gubbins into sanger-pathogens-master

Update fork with changes to Sanger master branch.

nickjcroucher committed:
Resolve conflicts before merging with original repository

nickjcroucher committed:
Update dependency installation script

nickjcroucher committed:
Apply patch gubbins_exe_in_usr_lib.patch

nickjcroucher committed:
Merge pull request #8 from nickjcroucher/patching

Apply patches from Debian version

nickjcroucher committed:
Update dockerfile

nickjcroucher committed:
Merge pull request #9 from nickjcroucher/docker

Update dockerfile to include RAxML-NG and rapidNJ

nickjcroucher committed:
Apply patch spelling-error-in-binary.patch

nickjcroucher committed:
Merge pull request #10 from nickjcroucher/patching

Apply patch spelling-error-in-binary.patch

nickjcroucher committed:
Update tree builder versions in Dockerfile

nickjcroucher committed:
Merge pull request #11 from nickjcroucher/docker

Update tree builder versions in Dockerfile

nickjcroucher committed:
Github actions for CI

nickjcroucher committed:
Temporary testing change

nickjcroucher committed:
Remove travis CI

nickjcroucher committed:
Add Github actions yml

nickjcroucher committed:
Reformat python test

nickjcroucher committed:
Define linux test

nickjcroucher committed:
Remove missing packages from environment.yml

nickjcroucher committed:
Change check installation

nickjcroucher committed:
Change check installation

nickjcroucher committed:
Remove old test command file

nickjcroucher committed:
Use sudo for installation of check

nickjcroucher committed:
Use autoreconf for check installation

nickjcroucher committed:
Install subunit as part of checks

nickjcroucher committed:
Fix to subunit installation

nickjcroucher committed:
Add cpuunit installation

nickjcroucher committed:
Change cpuunit installation

nickjcroucher committed:
Add cpuunit configuration

nickjcroucher committed:
Reorder installations

nickjcroucher committed:
Ensure C++ tests use defined, non-obsolete functions

nickjcroucher committed:
Edit ck_assert commands

nickjcroucher committed:
Add OSX test

nickjcroucher committed:
Change conda location in OSX test

nickjcroucher committed:
Add conda install automake autoconf libtool for OSX

nickjcroucher committed:
Add conda install pkg-config openssl libevent for OSX

nickjcroucher committed:
Add conda install berkeley-db4 boost boost-build for OSX

nickjcroucher committed:
Remove missing packages for conda install for OSX

nickjcroucher committed:
Use apt install for OSX

nickjcroucher committed:
Use homebrew install for OSX

nickjcroucher committed:
Add wheel as dependency

nickjcroucher committed:
Add wheel as a conda dependency

nickjcroucher committed:
Add wheel to setup.py

nickjcroucher committed:
Add pip install wheel to setup.py

nickjcroucher committed:
Add pip install scipy to setup.py

nickjcroucher committed:
Update python test scripts

nickjcroucher committed:
Use pytest

nickjcroucher committed:
Fix conda env name

nickjcroucher committed:
Change conda use for installation

nickjcroucher committed:
Change cppunit installation

nickjcroucher committed:
Change cppunit installation

nickjcroucher committed:
Change subunit configuration

nickjcroucher committed:
Change subunit cppunit flags

nickjcroucher committed:
Use installation from source

nickjcroucher committed:
Add dendropy as a dependency and change error reporting

nickjcroucher committed:
Additions to env

nickjcroucher committed:
Remove functools from env

nickjcroucher committed:
Change libcheck install prefix

nickjcroucher committed:
Change libcheck install prefix to conda env

nickjcroucher committed:
Add conda to make search path

nickjcroucher committed:
Reformat flags for libtools

nickjcroucher committed:
Restore cppunit source installation for OSX

nickjcroucher committed:
Change library search path

nickjcroucher committed:
Consistent install paths for OSX

nickjcroucher committed:
Add further flags for subunit

nickjcroucher committed:
Add further flags for subunit

nickjcroucher committed:
Add flags to gubbins configure

nickjcroucher committed:
Fix gubbins configure flags

nickjcroucher committed:
Change gubbins make flags for OSX

nickjcroucher committed:
Remove source installations from OSX

nickjcroucher committed:
Change flags to autoconfigure

nickjcroucher committed:
Change flags to autoconfigure for Linux

nickjcroucher committed:
Remove rendundancy in installation

nickjcroucher committed:
Edit flags for Linux installation

nickjcroucher committed:
Edit flags for cppunit Linux installation

nickjcroucher committed:
Restore cppunit Linux installation

nickjcroucher committed:
Restore cppunit installation

nickjcroucher committed:
Fix cppunit installation

nickjcroucher committed:
Remove unnecessary packages

nickjcroucher committed:
Prepare for merge with master

nickjcroucher committed:
Merge pull request #12 from nickjcroucher/ci_switch

Change CI to Github Actions

nickjcroucher committed:
Test how many errors are due to version not being established

nickjcroucher committed:
Remove obsolete tests

nickjcroucher committed:
Update expected file outputs

nickjcroucher committed:
Inactivate test for 3 sequences in alignment

nickjcroucher committed:
Update test tree builders

nickjcroucher committed:
Change tree_method file names

nickjcroucher committed:
Make test tree names consistent

nickjcroucher committed:
Update comparator for midpoint rooting tests

nickjcroucher committed:
Update bifurcation test result

nickjcroucher committed:
Move test file

nickjcroucher committed:
Pass extra arguments to IQtree builder in test

nickjcroucher committed:
Fix prefix for RAxML reconstruction

nickjcroucher committed:
Update package metadata

nickjcroucher committed:
Fix clean up test

nickjcroucher committed:
Fix to Debian bug #971111

nickjcroucher committed:
Remove pkg_resources dependency

nickjcroucher committed:
Update regex search strings

nickjcroucher committed:
Restore pkg_resources as dependency of subunit

nickjcroucher committed:
Fix environment.yml typo

nickjcroucher committed:
Restore working linux subunit installation

nickjcroucher committed:
Remove pkg_resources error

nickjcroucher committed:
Update linux installation commands

nickjcroucher committed:
Change imports into common.py

nickjcroucher committed:
Fix description function

nickjcroucher committed:
Change codecov installation on linux

nickjcroucher committed:
Change version handling

nickjcroucher committed:
Update codecov settings

nickjcroucher committed:
Add installation of pytest-cov

nickjcroucher committed:
Remove codecov command

nickjcroucher committed:
Move run_gubbins.py

nickjcroucher committed:
Fix raxmlng model fitting file names

nickjcroucher committed:
Remove unnecessary import

nickjcroucher committed:
Remove unneeded file

nickjcroucher committed:
Add model fitting tests

nickjcroucher committed:
Remove obsolete scripts directory

nickjcroucher committed:
Restructure runner script

nickjcroucher committed:
Change version retrieval

nickjcroucher committed:
Remove scripts dir reference from makefile

nickjcroucher committed:
Fix null pointer warning in test

nickjcroucher committed:
Bump version

nickjcroucher committed:
Change to python 3.x and fix ambiguous test

nickjcroucher committed:
Fix code coverage command for python tests

nickjcroucher committed:
Change coverage calculation for python code

nickjcroucher committed:
Change python testing to use consistent python version

nickjcroucher committed:
Change paths for testing files

nickjcroucher committed:
Change handling of file paths in tests

nickjcroucher committed:
Further fixes to test data paths

nickjcroucher committed:
Make tests give verbose outputs

nickjcroucher committed:
Extra clean up for tests

nickjcroucher committed:
Extra clean up for tests

nickjcroucher committed:
Make clean up test verbose

nickjcroucher committed:
Fix clean up test

nickjcroucher committed:
Test for RAxML-specific problems

nickjcroucher committed:
Exit when unable to find suitable RAxML software

nickjcroucher committed:
Define working directory

nickjcroucher committed:
Refine clean up function

nickjcroucher committed:
Fix clean up function

nickjcroucher committed:
Fix clean up function

nickjcroucher committed:
Multi-thread tests

nickjcroucher committed:
Update code for selecting RAxML algorithm based on CPU

nickjcroucher committed:
Simplify code coverage upload

nickjcroucher committed:
Change names of intermediate files

nickjcroucher committed:
Introduce bootstrapping tests

nickjcroucher committed:
Update help message

degreesKelvin commented:
> Happy new year! Addresses most of the improvements discussed in [nickjcroucher#1](https://github.com/nickjcroucher/gubbins/issues/1). CI switched to Github Actions and currently passing all tests on Linux and OSX. Just restructuring of the code and packaging to address, which seems easier to deal with post-PR.

Hi @nickjcroucher, I have installed Gubbins in Ubuntu based on the instructions in the manual, but it is still in the 1.2.4 version and I encountered memory resource issues. Can you please provide the instructions on how I can upgrade to the latest version of Gubbins?

Thank you and Happy New Year!

Kind Regards,
Kelvin

nickjcroucher commented:
Hi @degreesKelvin - I would not recommend using v3.0.0 yet, as we have not validated the accuracy of the results (coming soon hopefully!) - in the first instance, I would try installing the latest version available on conda (v2.4.1), which has some fixes to memory usage (it also uses marginal ancestral reconstruction, which may not be as accurate as the joint ancestral reconstruction in v1.2.4, but is likely to be faster and more memory efficient). If v2.4.1 doesn't work either, let me know - hopefully v3.0 will be ready for wider use soon!

degreesKelvin commented:
Thank you Nick!
It is showing the correct version now:

run_gubbins.py --version
2.4.1

Kind Regards,
Kelvin 😊


From: nickjcroucher <notifications@github.com>
Sent: Friday, 15 January 2021 4:59 PM
To: sanger-pathogens/gubbins <gubbins@noreply.github.com>
Cc: Kelvin Leong <kelvin.leong@utas.edu.au>; Mention <mention@noreply.github.com>
Subject: Re: [sanger-pathogens/gubbins] Upgrade to version 3.0 (#291)


Hi @degreesKelvin<https://github.com/degreesKelvin> - I would not recommend using v3.0.0 yet, as we have not validated the accuracy of the results (coming soon hopefully!) - in the first instance, I would try installing the latest version available on conda (v2.4.1), which has some fixes to memory usage (it also uses marginal ancestral reconstruction, which may not be as accurate as the joint ancestral reconstruction in v1.2.4, but is likely to be faster and more memory efficient). If v2.4.1 doesn't work either, let me know - hopefully v3.0 will be ready for wider use soon!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/sanger-pathogens/gubbins/pull/291#issuecomment-760763928>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AP3H2WDVVKKYH37MW52AGUTSZ77VPANCNFSM4VXIU7KQ>.


University of Tasmania Electronic Communications Policy (December, 2014).
This email is confidential, and is for the intended recipient only. Access, disclosure, copying, distribution, or reliance on any of it by anyone outside the intended recipient organisation is prohibited and may be a criminal offence. Please delete if obtained in error and email confirmation to the sender. The views expressed in this email are not necessarily the views of the University of Tasmania, unless clearly intended otherwise.

puethe commented on python/gubbins/common.py:
```diff
@@ -185,12 +207,16 @@ def parse_and_run(input_args, program_description=""):
-            os.chdir(current_directory)
+            if current_tree_builder == "star":
+                # Move star tree into temp dir
+                subprocess.check_call(["mv", tree_builder.tree_prefix + current_basename + tree_builder.tree_suffix, built_tree])
```
Better to use library (os/shutil)

nickjcroucher replied:
Thanks, good point - done in cf21522.

puethe commented on python/gubbins/common.py:
```diff
@@ -293,6 +438,149 @@ def parse_and_run(input_args, program_description=""):
+    elif algorithm_choice == "rapidnj":
+        initialised_algorithm = RapidNJ(threads = input_args.threads, model = model, bootstrap = input_args.bootstrap, verbose = input_args.verbose, additional_args = extra)
+    elif algorithm_choice == "star":
+        initialised_algorithm = Star()
```
no arguments passed to Star?

nickjcroucher replied:
It's as simple as that!

nickjcroucher committed:
Use shutil to move file

nickjcroucher committed:
Remove Makefile.in

nickjcroucher commented:
Makefile removed in 2377005.

nickjcroucher committed:
Specify versions in environment.yml

nickjcroucher committed:
Add missing dependencies to installation files

nickjcroucher committed:
Improve memory management error message

nickjcroucher committed:
Change menu options

nickjcroucher commented:
Further improvements suggested by @puethe:
•	Makefile.in gone in 2377005
•	Versions specified in yml in aa75c84
•	Added missing dependencies to setup.py and environment.yml in d8be9e1
•	Improved pyjar error message in 77883a8
•	Changed menu options in 51f3c46

nickjcroucher committed:
apt-get only run with Linux

nickjcroucher committed:
Remove downloaded files

nickjcroucher committed:
Make raxmlng test more stable

nickjcroucher committed:
Clarify menu options

nickjcroucher committed:
Remove version from __init__

nickjcroucher committed:
Add to path for package search

nickjcroucher committed:
Append to PATH before installation

nickjcroucher committed:
Edit threads argument description

nickjcroucher committed:
Merge version information

nickjcroucher committed:
Add make install to testing

nickjcroucher committed:
Try sudo install

nickjcroucher committed:
Change install prefix

nickjcroucher committed:
Tidy up installation and version parsing
